sympy/stats/crv.py

```diff
@@ -444,12 +444,17 @@ def probability(self, condition, **kwargs):
         except NotImplementedError:
             from sympy.stats.rv import density
             expr = condition.lhs - condition.rhs
-            dens = density(expr, **kwargs)
+            comp = 0
+            if len(random_symbols(expr)) == 0:
+                dens = self.density
+                comp = condition.rhs
+            else:
+                dens = density(expr, **kwargs)
```

Comment:
It would be better to set `comp = 0` in this branch rather than set it above the if.